core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/BPMBillingProjectDeleteRunnerSpec.scala

```diff
@@ -0,0 +1,464 @@
+      assert(message.get.contains(wsmExceptionMessage))
+      Future.successful(1)
+    }
+    val runner = spy(
+      new BPMBillingProjectDeleteRunner(
+        mock[SamDAO],
+        mock[GoogleServicesDAO],
+        wsmDao,
+        billingRepository,
+        mock[BillingProjectLifecycle]
+      )
+    )
+    doReturn(Future.successful(ctx)).when(runner).getUserCtx(ArgumentMatchers.eq(userEmail))(ArgumentMatchers.any())
+
+    whenReady(runner(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Complete)
+
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(DeletionFailed),
+      ArgumentMatchers.any[Some[String]]()
+    )
+  }
+
+  it should "reports a sensible message when no message is set in the error report" in {
+    val ctx = mock[RawlsRequestContext]
+    val wsmDao = mock[WorkspaceManagerDAO]
+    val landingZoneId = "7db89d7c-ef8b-4eaa-aef8-a62bd66cb095"
+    val monitorRecord: WorkspaceManagerResourceMonitorRecord = WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      JobType.AzureBillingProjectDelete,
+      None,
+      Some(billingProjectName.value),
+      Some(userEmail),
+      Timestamp.from(Instant.now())
+    )
+
+    val jobReport = new DeleteAzureLandingZoneJobResult()
+
+    when(
+      wsmDao.getDeleteLandingZoneResult(
+        ArgumentMatchers.eq(monitorRecord.jobControlId.toString),
+        ArgumentMatchers.eq(UUID.fromString(landingZoneId)),
+        ArgumentMatchers.any()
+      )
+    ).thenReturn(jobReport)
+
+    val billingRepository = mock[BillingRepository]
+    when(billingRepository.getLandingZoneId(ArgumentMatchers.eq(billingProjectName))(any()))
+      .thenReturn(Future.successful(Some(landingZoneId)))
+    when(
+      billingRepository.updateCreationStatus(
+        ArgumentMatchers.eq(billingProjectName),
+        ArgumentMatchers.eq(DeletionFailed),
+        ArgumentMatchers.any[Some[String]]()
+      )
+    ).thenAnswer { invocation =>
+      val message: Option[String] = invocation.getArgument(2)
+      assert(message.get.contains("deletion failed"))
+      assert(message.get.contains("no error"))
+      Future.successful(1)
+    }
+    val runner = spy(
+      new BPMBillingProjectDeleteRunner(
+        mock[SamDAO],
+        mock[GoogleServicesDAO],
+        wsmDao,
+        billingRepository,
+        mock[BillingProjectLifecycle]
+      )
+    )
+    doReturn(Future.successful(ctx)).when(runner).getUserCtx(ArgumentMatchers.eq(userEmail))(ArgumentMatchers.any())
+
+    whenReady(runner(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Complete)
+
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(DeletionFailed),
+      ArgumentMatchers.any[Some[String]]()
+    )
+  }
+
+  it should "continue running when the landing zone delete job is still running" in {}
```

Comment:
Looks like you forgot to come back to this test case.

Reply:
fixed